source/ui/dataman/AMScanView.cpp

```diff
@@ -412,7 +469,30 @@ void AMScanView::onScanAdded(AMScan *scan)
 
 	modeBar_->showSpectra_->setEnabled(!sources.isEmpty());
 	modeBar_->showSpectra_->setVisible(!sources.isEmpty());
+
+	// Apply changes to views.
+
 	spectrumView_->setDataSources(sources);
+
+	QString xUnits = "";
+
+	if (scan && scan->rawData() && scan->rawData()->scanAxesCount() >= 1) {
+		xUnits = scan->rawData()->scanAxisAt(0).units;
+	}
+
+	foreach (AMScanViewInternal *scanView, views_) {
+		if (scanView && scanView->tools()) {
```

Comment:
I agree that this code block is awkward and can benefit from some simplification. 

It's like this right now because 'units' is neither a property of the scan, the general AMScanView, nor of the AMScanViewInternal subclasses visualizing the scan data (AMScanViewInternal, AMScanViewExclusiveView, AMScanViewMultiView, etc.). The x axis units are a property of the first scan axis in the scan's raw data and the y axis units a property of each individual data source displayed (trickier to get at and manage, which is why y axis units aren't implemented now). 

The main goal of the work in this pull request is to display the data coordinates and units of a click position. Because I'd chosen to go the route of using the tools as models for the tool views, I've added a 'units' variable to MPlotDataPositionTool. So it's really just this particular tool that cares about the units (not any of the views, or other tools). Because there is no requirement that all AMScanViewInternal views use the same kinds of tools, a 'setUnits' type method in these classes would imply a relationship that presently isn't there and do nothing if that view didn't contain an MPlotDataPositionTool, which isn't really desirable either.

An alternative could be to make 'units' a property of the views, then a 'setUnits' type method would make a lot of sense. It might be worth the code refactor, either now or later on.